src/views/Users/ViewUser/index.jsx

```diff
@@ -0,0 +1,470 @@
+      </Grid>
+    </Grid>
+  );
+  const renderPermission = permission =>
+    zip(
+      [permission.name],
+      ((permission.options && permission.options.products) || []).concat([
+        'add',
+      ]),
+      ((permission.options && permission.options.actions) || []).concat(['add'])
+    ).map((row, index) => (
+      // TODO: need to replace key with something that's not the index, but also won't
+      // change when the data changes.
+      <Grid container spacing={2} key={index} className={classes.gridWithIcon}>
+        <Grid item xs>
+          {row[0] !== undefined && (
+            <TextField
+              required
+              value={row[0]}
+              className={classes.fullWidth}
+              disabled={permission.metadata.isAdditional ? false : !isNewUser}
+              onChange={e =>
+                handlePermissionNameChange(permission, index, e.target.value)
+              }
+            />
+          )}
+        </Grid>
+        {row[1] === undefined && <Grid item xs />}
+        {row[1] !== undefined && row[1] !== 'add' && (
+          <Fragment>
+            <Grid item xs>
+              <TextField
+                value={row[1]}
+                style={{ width: '85%' }}
+                onChange={e =>
+                  handleProductNameChange(permission, index, e.target.value)
+                }
+              />
+              <IconButton
+                onClick={() => handleProductDelete(permission, index)}
+                style={{ width: '15%' }}>
+                <DeleteIcon />
+              </IconButton>
+            </Grid>
+          </Fragment>
+        )}
+        {row[1] === 'add' && (
+          <Grid item xs className={classes.addGrid}>
+            <Button
+              onClick={() => handleProductAdd(permission)}
+              className={classes.fullWidth}
+              variant="outlined">
+              <PlusIcon />
+            </Button>
+          </Grid>
+        )}
+        {row[2] === undefined && <Grid item xs />}
+        {row[2] !== undefined && row[2] !== 'add' && (
+          <Fragment>
+            <Grid item xs>
+              <TextField
+                value={row[2]}
+                onChange={e =>
+                  handleActionNameChange(permission, index, e.target.value)
+                }
+                style={{ width: '85%' }}
+              />
+              <IconButton
+                onClick={() => handleActionDelete(permission, index)}
+                style={{ width: '15%' }}>
+                <DeleteIcon />
+              </IconButton>
+            </Grid>
+          </Fragment>
+        )}
+        {row[2] === 'add' && (
+          <Grid item xs className={classes.addGrid}>
+            <Button
+              className={classes.fullWidth}
+              variant="outlined"
+              onClick={() => handleActionAdd(permission)}>
+              <PlusIcon />
+            </Button>
+          </Grid>
+        )}
+      </Grid>
+    ));
```

Comment:
For the permissions, I think we should be able to use the multi selection feature of our `AutoCompleteText` component.

Example: https://codesandbox.io/s/github/kentcdodds/downshift-examples/tree/master/?module=%2Fsrc%2Fordered-examples%2F04-multi-select.js&moduleview=1
Docs: https://github.com/downshift-js/downshift

_Note: `src/components/AutoComplete` is merely a wrapper around the Downshift package that makes the text field have a material look._

Reply:
I may be misunderstanding what you're suggesting, but I'm not sure multiselect is a great idea here. Not all permissions support the same product and/or action restrictions. And where they do, we don't always grant them the same to some accounts. For example, one account is production has the "rule" permission with the "modify" action restriction, but also the "release" permission with the "create" and "modify" action restrictions.

Reply:
Is the stuff we're trying to do here different than the multi select text fields in the old UI:

![Screen Shot 2019-07-05 at 10 56 00 AM](https://user-images.githubusercontent.com/3766511/60730637-9c7acf80-9f13-11e9-949a-8781b9974563.png)

Reply:
Oh, my bad! I thought you were talking about using multiselect for the permission name. Let's do this!